app/views/layouts/application.haml

```diff
@@ -66,7 +64,9 @@
       report_analytics_event.apply(null, #{@analytics_event});
   - if @gtm_datalayer
     :javascript
-      gtm_datalayer_push(#{@gtm_datalayer.to_json});
+      if(window.ST != null && typeof window.ST.gtmPush === 'function') {
```

Comment:
@mporkola, In this particular case, it wouldn't matter. The use case for ST is defined clearly as object literal.

However, checking just existence of variable (in if statement) is also taking into account empty string, false, 0, NaN etc. 
So, in general, it is a better practice to just check null & undefined (with "!= null"), when the purpose is to check whether it is unintentionally undefined or explicitly defined to null by devs. 
To me this is more like a best practice thing: check just the existence in 'if' statements and don't assume anything about what might be assigned inside of dynamic variables.